source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -157,6 +159,8 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
   route_config_provider_ = Router::RouteConfigProviderUtil::create(config, context_, stats_prefix_,
                                                                    route_config_provider_manager_);
 
+  http_parser_set_max_header_size(max_request_headers_kb_ * 1024);
```

Comment:
I don't want to leak this into the HCM code. So we determined this is setting a global variable? That is really unfortunate and I don't see how we can sanely support this with different configurations. IMO we need to set some global limit for this. Compile time define for Envoy? CLI flag? Global setting? This would be the outer limit of what we can support at the per-listener config layer. Thoughts on this?

Reply:
I thought about this more, and can we potentially just not use this library setting? We can just implement something like the HTTP/2 codec saveHeaders check in the HTTP/1 codec. Since the HTTP/1 parser is streaming, it should be pretty trivial to do our own check?